### PR TITLE
Add the name of a solid next to it in tree

### PR DESCRIPTION
**Pull request by ShuffleWire**
Fix https://github.com/cyberbotics/webots/issues/3157

However, the fix is not the right way to do it, but I don't which one is the good one : 
- add "Solid" in either WbNodeUtilities::isDeviceTypeName or WbNodeUtilities::isSolidDeviceTypeName
- or remove the condition that this PR modify. In fact I don't understand why should we limit the display of the name to certains types. As long as the node have a "name" field, we should be good to display it.

### CODE REVIEW

**Comment by omichel** [comment]
Thank you for this contribution.
I believe the second option you propose is the best: as long as the node has a "name" field (and no DEF name), we should use it to display it in the scene tree.

**Comment by omichel** [comment]
It would be nice to also add a line in the change log about this.

**Comment by ShuffleWire** [comment]
I'm not convinced if we should add the name for node which are def or use, however.

**Comment by omichel** [comment]
USE/DEF name are displayed if specifed, this already gives a good indication on what the node is (if the USE/DEF name is well chosen). So, I would not display the "name" field in case the DEF/USE name is specified, as it would provide too much information which may be confusing.

**Comment by ad-daniel** [comment]
Just a heads up, when the PR already has a reviewer assigned, re-request the review directly to that person after making the necessary changes. This way it's less likely to go unnoticed. Here's an example
rev

**Comment by ShuffleWire** [comment]
> Just a heads up, when the PR already has a reviewer assigned, re-request the review directly to that person after making the necessary changes. This way it's less likely to go unnoticed. Here's an example

Thank for the advice :) 
However, afaik, I do think I can't to much more from my side : 
image
omichel is already requested, isn't ? and I don't think their are any pending requested change

File: src/webots/scene_tree/WbTreeItem.cpp [hunk 1]
```diff
@@ -129,11 +129,9 @@ QString WbTreeItem::data() const {
     case NODE: {
       QString fullName = mNode->fullName();
       if (!fullName.startsWith("DEF ") && !fullName.startsWith("USE ")) {
-        if (WbNodeUtilities::isDeviceTypeName(mNode->nodeModelName())) {
-          WbSFString *name = mNode->findSFString("name");
-          if (name)
-            fullName += " \"" + name->value() + "\"";
-        }
+        WbSFString *name = mNode->findSFString("name");
```
Comment (omichel) [inline src/webots/scene_tree/WbTreeItem.cpp:132]:
```suggestion
        const WbSFString *name = mNode->findSFString("name");
```

**Review by omichel — CHANGES_REQUESTED** [review]
That looks very good to me.
Maybe we can just take advantage of this PR to make it const correct?

File: src/webots/scene_tree/WbTreeItem.cpp (hunk 1 above)
Comment (ShuffleWire) [reply src/webots/scene_tree/WbTreeItem.cpp]: of course !

**Review by omichel — APPROVED** [review]
Thank you.

**Comment by ad-daniel** [comment]
> Thank for the advice :)
> However, afaik, I do think I can't to much more from my side :

rev
We generally handle the assignment of the reviewer, what I mean is after doing the changes and going out of draft, you can use this button to re-request the specific person for a second review.

**Comment by ShuffleWire** [comment]
@ad-daniel ok so going in/out of draft is the trick. Will be trying that next time !

**Pull request merged**